source/s2n/s2n_tls_channel_handler.c

```diff
@@ -924,6 +950,48 @@ static int s2n_monotonic_clock_time_nanoseconds(void *context, uint64_t *time_in
     return 0;
 }
 
+static int s_tls_ctx_pkcs11_setup(struct s2n_ctx *s2n_ctx, const struct aws_tls_ctx_options *options) {
+    /* anything initialized in this function is cleaned up during s_s2n_ctx_destroy()
+     * so don't worry about cleaning up unless it's some tmp heap allocation */
+
+    s2n_ctx->pkcs11.lib = aws_pkcs11_lib_acquire(options->pkcs11.lib);
```

Comment:
What happens when this function fails?

Reply:
this function cannot fail. It's just a atomic increment, that does a "return this" at the end, for the sake of ergonomics